src/main/java/com/larscheidschmitzhermes/nexus3/github/oauth/plugin/api/GithubApiClient.java

```diff
@@ -94,45 +90,35 @@ private GithubPrincipal doAuthz(String loginName, char[] token) throws GithubAut
         return principal;
     }
 
-    private GithubUser retrieveGithubUser(String loginName, char[] token) throws GithubAuthenticationException {
-        try {
-            HttpGet userRequest = new HttpGet(configuration.getGithubUserUri());
-            userRequest.addHeader(constructGithubAuthorizationHeader(token));
-            HttpResponse userResponse = client.execute(userRequest);
 
-            if (userResponse.getStatusLine().getStatusCode() != 200) {
-                LOGGER.warn("Authentication failed, status code was {}",
-                        userResponse.getStatusLine().getStatusCode());
-                userRequest.releaseConnection();
-                throw new GithubAuthenticationException("Authentication failed.");
-            }
+    private GithubUser retrieveGithubUser(String loginName, char[] token) throws GithubAuthenticationException {
+        try (InputStreamReader reader = executeGet(configuration.getGithubUserUri(), token)) {
```

Comment:
I still don't like the giant try catch around basically the whole method. can you make that smaller so that it only covers the relevant lines or even better move it to it's own methods that is throwing a `GithubAuthenticationException` itself?

Reply:
Agree, the try is there to close the reader on exception and should only surround the mapping part.